crates/wysiwyg/src/composer_model/replace_text.rs

```diff
@@ -176,6 +188,54 @@ where
         self.create_update_replace_all()
     }
 
+    fn set_new_text_in_sibling_node(
+        &mut self,
+        node_handle: DomHandle,
+        new_text: S,
+    ) {
+        if let Some(sibling_text_node) =
+            self.first_sibling_text_node_mut(&node_handle)
+        {
+            let mut data = sibling_text_node.data().to_owned();
+            data.insert(0, &new_text);
+            sibling_text_node.set_data(data);
+        } else if !new_text.is_empty() {
+            let new_child = DomNode::new_text(new_text);
+            let parent = self.state.dom.parent_mut(&node_handle);
+            let index = node_handle.index_in_parent() + 1;
+            parent.insert_child(index, new_child);
+        }
+    }
+
+    fn first_sibling_text_node_mut(
+        &mut self,
+        node_handle: &DomHandle,
+    ) -> Option<&mut TextNode<S>> {
+        let parent = self.state.dom.parent(node_handle);
+        let children_number = parent.children().len();
+        if node_handle.index_in_parent() < children_number - 1 {
+            let sibling =
+                self.state.dom.lookup_node_mut(&node_handle.next_sibling());
+            if let DomNode::Text(sibling_text_node) = sibling {
+                Some(sibling_text_node)
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    }
+
+    fn first_link_node_handle(&self, range: &Range) -> Option<DomHandle> {
```

Comment:
`fn find_link_node_to_shrink_at_end`? I can't find a good name for this function. @andybalaam , any ideas?